% Generated by roxygen2: do not edit by hand
% Please edit documentation in R/getLincRNAPvalue_cluster.R
\name{getLincRNAPvalueWithPreFilter}
\alias{getLincRNAPvalueWithPreFilter}
\title{Run CNCDriver lincRNA p-value calculation}
\usage{
getLincRNAPvalueWithPreFilter(inputFileDir, outputFileDir,
  lincRNARegionBedFile, elementKeyWord = "lincRNA", minPoints = 2,
  dRadius = 50, triNucleotideDistributionFile,
  filterOutBlacklistMutations, mutationBlacklistFile,
  replicationTimingGenomeBinnedFile, replicationTimingElementBinnedFile,
  tumorType, mutationType, cellType, replicationTimingCutOff,
  seedNum = 42, reSampleIterations, reRunPvalueCutOff = 0.1, useCores,
  taskNum, unitSize, debugMode = FALSE)
}
\arguments{
\item{inputFileDir}{The path for prepared R object file [for example,reducedFunseqOutputNCDS_GBM.Rd]}

\item{outputFileDir}{The path for output files}

\item{lincRNARegionBedFile}{The defintion of promoter regions in bed file format}

\item{elementKeyWord}{Default is "lincRNA", the keyword of lincRNA annotation in FunSeq2}

\item{minPoints}{default is 2 [ unit: samples ]}

\item{dRadius}{default is 50 [ unit: bp ]}

\item{triNucleotideDistributionFile}{Cancer type specific mutation counts in 96 trinucleotide category}

\item{filterOutBlacklistMutations}{TRUE or FALSE}

\item{mutationBlacklistFile}{The file for mutation blacklist regions}

\item{replicationTimingGenomeBinnedFile}{Replication timing value at 1MB bin}

\item{replicationTimingElementBinnedFile}{Replication timing value of each element within 1MB bin}

\item{tumorType}{Study name}

\item{mutationType}{User provided mutated gene list}

\item{cellType}{Cell type of the study, [ BJ, GM12878, HeLaS3, HepG2, IMR90, K562, MCF7, SKNSH or average ]}

\item{replicationTimingCutOff}{Default is 0.2, a numeric value ranging from 0 to 1}

\item{seedNum}{User provided random number seed, default is 42}

\item{reSampleIterations}{User provided re-sapling iteration numbers}

\item{reRunPvalueCutOff}{Default is 0.1, a numeric value ranging from 0 to 1}

\item{useCores}{Default is 1, number of cpu to use}

\item{taskNum}{Default is 0, 0 means to run all elements}

\item{unitSize}{Number of elements to run per task}

\item{debugMode}{TRUE or FALSE}
}
\value{
results data frame
}
\description{
Run CNCDriver lincRNA p-value calculation
}
\examples{
#date<-getRunDates(latest=TRUE)
cancerType<-"KIRC"
selectedSampleId<-NA
#worDir<-getwd()
mutSig2CVthreshold<-0.1
rareMutationUpperLimit<-0.3
rareMutationLowerLimit<-0.1
rareMutationFreq<-0.02

#runNetBox2(dataDir,cancerType,
#           mutationList,ampGeneList,delGeneList,epiSilencedList,
#           mutationFreq,ampGeneFreq,delGeneFreq,epiSilencedFreq,
#           pathwayCommonsDb,directed,
#           linkerPValThreshold,communityDetectionMethod,
#           keepIsolatedNodes,verbose=TRUE)

}
\concept{CNCDriver}
